### PR TITLE
Add support for listing metadata

### PR DESCRIPTION
**Pull request by kirre-bylund**
Also a bunch of drive by improvements and fixes:
- [feat: Add parsing of Json Array from String to Utilities](https://github.com/lootlocker/unreal-sdk/commit/aaf252372d6849512b33e81d8131aa45faee9ef4)
- [fix: Make JsonObject ptr null if serialization fails](https://github.com/lootlocker/unreal-sdk/commit/7ba6b027b24e9621089e97143e897d08e273dade)
- [fix: Move GoogleSubsystemHelper methods to same category as standard](https://github.com/lootlocker/unreal-sdk/commit/cce5fdff6d60232b5468e04d0c33db1c705d0541)
- [fix: Document undocumented properties in LeaderboardDetails](https://github.com/lootlocker/unreal-sdk/commit/bab6e6bf34ea457a8c9c2b6d338720aa5ce3adce)
- [chore: Document pagination object fields](https://github.com/lootlocker/unreal-sdk/commit/6610afb5bf81a115428c31315203802ccad6ad3a)
- [chore: Correct parameter documentation for Feedback methods](https://github.com/lootlocker/unreal-sdk/commit/9de92ee78e67d25179086dae52b9fdda99283317)
- [feat: Add successful request logging as VeryVerbose](https://github.com/lootlocker/unreal-sdk/commit/0957fbfc81f9d8d215482d33ba40cadf2748a38b)
- [feat: Add Json Array to String conversion to Utilities](https://github.com/lootlocker/unreal-sdk/commit/6754ab6fd14aee8949fc99b42cefb21cbb5a0496)

### CODE REVIEW

**Review by AlmightyMikkel — APPROVED** [review]
Wow, Just wow. Incredible job Erik! This could not have been easy I am really impressed!

File: LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    LOOTLOCKERSDK_API bool TryGetValueAsInteger(int& Output) const;
+    /*
+     Get the value as a boolean. Returns true if value could be parsed in which case Output contains the bool, returns false if parsing failed which can happen if the string is not a convertible to a boolean (those are for example "0", "1", "true", "False", "yes", "NO", etc).
+     */
+    LOOTLOCKERSDK_API bool TryGetValueAsBool(bool& Output) const;
+    /*
+     Get the value as an unparsed json value. Returns true if value could be found in which case Output contains the JsonValue, returns false if the value field was not present.
+     */
+    LOOTLOCKERSDK_API bool TryGetRawValue(TSharedPtr<FJsonValue>& Output) const;
+    LOOTLOCKERSDK_API bool TryGetSerializedValue(FString& Output) const;
```
Comment (kirre-bylund) [inline LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h:122]: Missing documentation

File: LootLockerSDK/Source/LootLockerSDK/Private/GameAPI/LootLockerMetadataRequestHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2021 LootLocker
+
+#include "GameAPI/LootLockerMetadataRequestHandler.h"
+#include "Dom/JsonObject.h"
```
Comment (kirre-bylund) [inline LootLockerSDK/Source/LootLockerSDK/Private/GameAPI/LootLockerMetadataRequestHandler.cpp:4]: Is this good?

File: LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h (hunk 1 above)
Comment (JohannesLoot) [inline LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h:122]: Undocumented

File: LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    /*
+     Get the value as an unparsed json value. Returns true if value could be found in which case Output contains the JsonValue, returns false if the value field was not present.
+     */
+    template<typename T>
```
Comment (JohannesLoot) [inline LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h:126]: Intentional?

Comment (kirre-bylund) [reply LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h:126]: What? That I use a template? Yes

**Review by JohannesLoot — APPROVED** [review]
Only thing is the missing documentation on TryGetSerializedValue that it looks like we added at the same time :), so you're already on top of that.
Really really great work ⭐

File: LootLockerSDK/Source/LootLockerSDK/Private/GameAPI/LootLockerMetadataRequestHandler.cpp (hunk 2 above)
Comment (kirre-bylund) [reply LootLockerSDK/Source/LootLockerSDK/Private/GameAPI/LootLockerMetadataRequestHandler.cpp:4]: It is the Unreal Json object so it's fine

File: LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h (hunk 1 above)
Comment (kirre-bylund) [reply LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h:125]: Fixed

Comment (kirre-bylund) [reply LootLockerSDK/Source/LootLockerSDK/Public/GameAPI/LootLockerMetadataRequestHandler.h:125]: Fixed

**Pull request merged**